spec/Plugin/QueryDefaultsPluginSpec.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace spec\Http\Client\Common\Plugin;
+
+use Http\Client\Common\Plugin;
+use Psr\Http\Message\RequestInterface;
+use PhpSpec\ObjectBehavior;
+use Psr\Http\Message\UriInterface;
+
+class QueryDefaultsPluginSpec extends ObjectBehavior
+{
+    public function let()
+    {
+        $this->beConstructedWith([]);
+    }
+
+    public function it_is_initializable()
+    {
+        $this->shouldHaveType('Http\Client\Common\Plugin\QueryDefaultsPlugin');
+    }
+
+    public function it_is_a_plugin()
+    {
+        $this->shouldImplement(Plugin::class);
```

Comment:
This will break php 5.4 compatibility. Even though I think we should drop 5.4 support, this is not a good enough reason. 

FYI @sagikazarmark